virttest/utils_net.py

```diff
@@ -3390,17 +3390,17 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
         try:
             if linklocal:
                 return [a['addr'] for a in addr
-                        if a['addr'].lower().startswith(linklocal_prefix)][0]
+                        if a['addr'].lower().startswith(linklocal_prefix)][0].split("%")[0]
```

Comment:
The current IPv6 address is in the correct format.
```python
>>> netifaces.ifaddresses("switch").get(netifaces.AF_INET6)
[{'addr': '2620:52:0:10c8:5e:c2cc:ece7:ae26', 'netmask': 'ffff:ffff:ffff:ffff::/64'}, {'addr': 'fe80::6946:86f8:e70d:8c7c%switch', 'netmask': 'ffff:ffff:ffff:ffff::/64'}]
```
IPv6 addresses with the prefix `fe80::/64` are link-local addresses that are constructed by combining that prefix with the hardware address of the network device, so the device ID should be reserved. And you can see the return format of `ping6` also include it.
```bash
# ping6 fe80::6946:86f8:e70d:8c7c%switch -c 4
PING fe80::6946:86f8:e70d:8c7c%switch(fe80::6946:86f8:e70d:8c7c%switch) 56 data bytes
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=1 ttl=64 time=0.017 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=2 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=3 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=4 ttl=64 time=0.006 ms

# ping6 fe80::6946:86f8:e70d:8c7c -c 4
PING fe80::6946:86f8:e70d:8c7c(fe80::6946:86f8:e70d:8c7c) 56 data bytes
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=1 ttl=64 time=0.017 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=2 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=3 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=4 ttl=64 time=0.006 ms
```

Reply:
1. tested it under uefi shell, it doesn't work.

FS0:\> ping6 -n 5 -s fe80::9834:39ff:fe31:a413 fe80::af43:41fd:5437:9523%switch 

Ping6: Invalid IP6 address, fe80::af43:41fd:5437:9523%switch
 
2. remove \"'%switch \", it works well. I will double check with edk2 developer whether fix it under uefi shell. thanks

FS0:\> ping6 -n 5 -s fe80::9834:39ff:fe31:a413 fe80::af43:41fd:5437:9523 
Ping fe80::af43:41fd:5437:9523 16 data bytes
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=1 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=2 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=3 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=4 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=5 ttl=64 time0~9ms

5 packets transmitted, 5 received, 0% packet loss, time 0ms

Rtt(round trip time) min=0~9ms max=0~9ms avg=0~9ms